plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -206,16 +206,31 @@ func buildControllerRoles() ([]rbacv1.ClusterRole, []rbacv1.ClusterRoleBinding)
 		},
 	})
 	addControllerRole(&controllerRoles, &controllerRoleBindings, rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + "node-controller"},
+		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + "node-lifecycle-controller"},
```

Comment:
this will orphan the existing clusterrole on upgraded clusters? what is the reason for the rename?

Reply:
Ah, was wondering what happens there. So my justification for this is:
1) this is confusing cause the controller was always called `node-lifecycle` and not just `node`. 
2) `node-controller` has access to delete nodes, so leaving this would give it access to delete nodes even though it no longer needs it. 

I'm not sure what the implications of orphaned cluster roles are so I'll defer to your judgment here (worse case we can keep the `node-controller` role).  Any reason why we can't support a way to clean up orphaned controllers? Would that not work because of version skew between components?

Reply:
> Any reason why we can't support a way to clean up orphaned controllers? Would that not work because of version skew between components?

We take an extremely conservative position on permissions management and upgrades. Roles and permissions are never automatically removed or tightened by an upgrade, to ensure that no component that is currently working is ever broken by an upgrade removing its permission. 

New clusters would only have the current set of default roles/bindings

Reply:
Good to know! If orphaning the `node-controller` role is still an option I'd like to go down that path. Otherwise, we can continue to use the `node-controller` role here unchanged for the node lifecycle controller.

Reply:
Of these, I'd favor Option 2 so we don't leave an unused role floating around.

Reply:
> this is confusing cause the controller was always called node-lifecycle and not just node

For histoy's sake, it was actually called the `node` controller until @cheftako split things up in https://github.com/kubernetes/kubernetes/pull/57492